sympy/solvers/inequalities.py

```diff
@@ -481,11 +483,32 @@ def valid(x):
         if valid(pt):
             sol_sets.append(Interval(start, end, True, True))
 
+        elif solns == []: # if no solutions for equation e
+            nsol_sets = _real_conditions(expr,gen)
+        cv = Intersection(*nsol_sets).subs(gen,_gen)
         rv = Union(*sol_sets).subs(gen, _gen)
-
+        rv = Union(rv, cv)
     return rv if not relational else rv.as_relational(_gen)
 
 
+def _real_conditions(expr, gen):
+    from sympy.core import Mul, Pow
```

Comment:
@sampadsaha5 , thanks. Can you give your review on the whole PR so I can make changes if any?